### PR TITLE
Release checklist GH issue template

### PR DESCRIPTION
**Pull request by edwardchalstrey1**
## :white_check_mark: Checklist

- [x] You have given your pull request a meaningful title (_e.g._ `Enable foobar integration` rather than `515 foobar`).
- [x] You are targeting the appropriate branch. If you're not certain which one this is, it should be **`develop`**.
- [x] Your branch is up-to-date with the **target branch** (it probably was when you started, but it may have changed since then).
- [ ] You have marked this pull request as a **draft** and added `'[WIP]'` to the title if needed (if you're not yet ready to merge).
- [ ] You have formatted your code using appropriate automated tools (for example `./tests/AutoFormat_Powershell.ps1 -TargetPath <path to file or directory>` for Powershell).

### :arrow_heading_up: Summary

- Adds a template GH issue for tracking the testing of release candidates https://github.com/alan-turing-institute/data-safe-haven-team/issues/21
- Mentions this in the README

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release_checklist.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+-->
+
+Refer to the [Deployment](https://data-safe-haven.readthedocs.io/en/latest/deployment) section of our documentation when completing these steps.
+
+- [ ] Consult the `data-safe-haven/VERSIONING.md` guide and determine the version number of the new release. Record it in the title of this issue.
+- [ ] Create a release branch called e.g. `release-v0.0.1`
+- [ ] Draft a changelog for the release similar to our previous releases, see https://github.com/alan-turing-institute/data-safe-haven/releases
+- [ ] Deploy an SHM from this branch and save a transcript of the deployment logs in a secure location
+- [ ] Build an SRE compute image (SRD) and save transcripts of the logs in a secure location
+- Using the new image, deploy two SREs covering tier 2/3 and Guacamole/RDS
```
Comment (jemrobinson) [inline .github/ISSUE_TEMPLATE/release_checklist.md]:
```suggestion
- Using the new image, deploy two SREs which between them cover tiers 2 and 3 and Guacamole/Microsoft RDS
```

File: .github/ISSUE_TEMPLATE/release_checklist.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    - [ ] Save transcript of T2 SRE deplpoyment log to a secure location
```
Comment (jemrobinson) [inline .github/ISSUE_TEMPLATE/release_checklist.md:23]:
```suggestion
    - [ ] Save transcript of T2 SRE deployment log to a secure location
```

File: .github/ISSUE_TEMPLATE/release_checklist.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    - [ ] Save transcript of T3 SRE deplpoyment log to a secure location
```
Comment (jemrobinson) [inline .github/ISSUE_TEMPLATE/release_checklist.md]:
```suggestion
    - [ ] Save the transcript of your tier 2 SRE deployment
    - [ ] Save the transcript of your tier 3 SRE deployment
```

File: .github/ISSUE_TEMPLATE/release_checklist.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+- [ ] Complete the [Security evaluation checklist](https://data-safe-haven.readthedocs.io/en/latest/deployment/security_checklist.html) from the deployment documentation
+
+**For MAJOR releases:**
+
+- [ ] A third party has carried out a full penetration test evaluating (1) external attack surface, (2) ability to exfiltrate data from the system, (3) ability to transfer data between SREs, (4) ability to escalate privileges on the SRD.
```
Comment (jemrobinson) [inline .github/ISSUE_TEMPLATE/release_checklist.md]:
```suggestion
- [ ] Confirm that a third party has carried out a full penetration test evaluating:
  1. external attack surface
  1. ability to exfiltrate data from the system
  1. ability to transfer data between SREs
  1. ability to escalate privileges on the SRD.
```

File: README.md [hunk 5]
```diff
@@ -41,6 +41,8 @@ Follow the link from any release to view and clone this repository as at that re
 
 Read our [versioning scheme](VERSIONING.md) for how we number and label releases, as well as details of releases that have been used in production and releases that have undergone formal security evaluation.
 
+When making a new release, open an issue oh GitHub and choose the `Release checklist` template, which can be used to track the completion of security checks for the release.
```
Comment (jemrobinson) [inline README.md]:
```suggestion
When making a new release, open an issue on GitHub and choose the `Release checklist` template, which can be used to track the completion of security checks for the release.
```

**Review by jemrobinson — CHANGES_REQUESTED** [review]
Mostly good. A couple of minor typos.

File: .github/ISSUE_TEMPLATE/release_checklist.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+Before reporting a problem please check the following. Replace the empty checkboxes [ ] below with checked ones [x] accordingly.
+-->
+
+Refer to the [Deployment](https://data-safe-haven.readthedocs.io/en/latest/deployment) section of our documentation when completing these steps.
+
+- [ ] Consult the `data-safe-haven/VERSIONING.md` guide and determine the version number of the new release. Record it in the title of this issue.
+- [ ] Create a release branch called e.g. `release-v0.0.1`
+- [ ] Draft a changelog for the release similar to our previous releases, see https://github.com/alan-turing-institute/data-safe-haven/releases
+- [ ] Deploy an SHM from this branch and save a transcript of the deployment logs in a secure location
+- [ ] Build an SRE compute image (SRD) and save transcripts of the logs in a secure location
```
Comment (edwardchalstrey1) [inline .github/ISSUE_TEMPLATE/release_checklist.md]:
```suggestion
- [ ] Build an SRE compute image (SRD) and save transcripts of the logs
```

File: .github/ISSUE_TEMPLATE/release_checklist.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+<!--
+Before reporting a problem please check the following. Replace the empty checkboxes [ ] below with checked ones [x] accordingly.
+-->
+
+Refer to the [Deployment](https://data-safe-haven.readthedocs.io/en/latest/deployment) section of our documentation when completing these steps.
+
+- [ ] Consult the `data-safe-haven/VERSIONING.md` guide and determine the version number of the new release. Record it in the title of this issue.
+- [ ] Create a release branch called e.g. `release-v0.0.1`
+- [ ] Draft a changelog for the release similar to our previous releases, see https://github.com/alan-turing-institute/data-safe-haven/releases
+- [ ] Deploy an SHM from this branch and save a transcript of the deployment logs in a secure location
```
Comment (edwardchalstrey1) [inline .github/ISSUE_TEMPLATE/release_checklist.md]:
```suggestion
- [ ] Deploy an SHM from this branch and save a transcript of the deployment logs
```

**Review by jemrobinson — APPROVED** [review]
LGTM

**Pull request merged**